e3sm_to_cmip/util.py

```diff
@@ -306,195 +359,389 @@ def print_var_info(handlers, freq=None, inpath=None, tables=None, outpath=None):
-        return table_path.name, False
-    else:
-        return table_path.name, True
+    return all_handlers
 
-def get_table_info(tables, table):
-    table = Path(tables, table)
-    if not table.exists():
-        raise ValueError(f"CMIP6 table doesnt exist: {table}")
-    with open(table, 'r') as instream:
-        return json.load(instream)
 
-def get_table_freq(table, freq):
-    if table == "CMIP6_Amon.json":
-        return f"CMIP6_{freq}.json"
-    elif table in ["CMIP6_Lmon.json", "CMIP6_LImon.json"] and freq != "mon":
-        return None
-    elif 'fx' in table and freq != "mon":
-        return None
-     
-    return f"{table[:-8]}{freq}.json"
+def _get_default_handlers() -> Dict[str, Dict[str, str]]:
+    """Gets the information for all default handlers in a nested dictionary.
 
+    The parent key is the CMIP name of the variable, and the value is a nested
+    dictionary storing key/value pairs for "name", "units", "table",
+    "method", "raw_variables", and "unit_conversion" (optional).
 
+    Example output:
+    ---------------
+    {
+        "abs550aer": {
+            "name": "abs550aer",
+            "units": "1",
+            "table": "CMIP6_AERmon.json",
+            "method": <function e3sm_to_cmip.default.default_handler(infiles, tables, user_input_path, **kwargs)>
+            "raw_variables": ["AODABS"],
+        },
+    }
 
-def load_handlers(handlers_path, var_list, tables, freq="mon", realm='atm', simple=False, debug=None):
+    Returns
+    -------
+    Dict[str, Dict[str, str]]
+        A dictionary of dictionaries, with each dictionary storing the
+        information for a default handler.
     """
-    load the cmor handler modules
+    # FIXME: Imports should be at the module level. However, moving this to the
+    # module level results in a circular import.
+    from e3sm_to_cmip.default import default_handler
 
-    Params:
+    resources_path = os.path.split(os.path.abspath(resources.__file__))[0]
+    defaults_path = os.path.join(resources_path, "default_handler_info.yaml")
+
+    with open(defaults_path, "r") as infile:
+        default_handlers = yaml.load(infile, Loader=yaml.SafeLoader)
+
+    # Convert the list of dicts to a nested dict, with the key being the
+    # "name" (cmip variable name).
+    default_handlers = {handler["cmip_name"]: handler for handler in default_handlers}
+
+    # Post-process the loaded default handlers to align with the example
+    # dictionary structure.
+    for key in default_handlers.keys():
+        # Add a "method" key with a value of default_handler to each nested dict.
+        default_handlers[key]["method"] = default_handler
+        # The "raw_variables" entry should be a list of strings.
+        default_handlers[key]["raw_variables"] = [
+            default_handlers[key].pop("e3sm_name")
+        ]
+        # Rename "cmip6_name" key to "name"
+        default_handlers[key]["name"] = default_handlers[key].pop("cmip_name")
+
+    return default_handlers
```

Comment:
This function reads the `default_handlers_info.yaml` file to create dict objects representing default handlers.